### PR TITLE
feat: Use Cachito to build UDI golang dependency gopls v0.7.2

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

re: https://issues.redhat.com/browse/CRW-3103

see also https://github.com/devspaces-samples/golang-health-check/pull/1

### CODE REVIEW

File: devspaces-udi/Dockerfile [hunk 1]
```diff
@@ -180,12 +182,12 @@ RUN \
     # Golang (Tech Preview)
     ########################################################################
     dnf -y -q install golang glibc-devel zlib-devel libstdc++ libstdc++-devel && \
-    mkdir -p /projects/.che/gopath /.cache ${HOME}/go && \
-    tar -xvf /tmp/assets/asset-golang-$(uname -m).tar.gz --strip-components=1 -C /projects/.che/gopath/ && \
-    for f in "/home/user" "/etc/passwd" "/etc/group" "/projects" "/.cache" "/usr/share/gocode" "/opt/app-root/src/"; do \
-        chgrp -R 0 ${f} && \
-        chmod -R g+rwX ${f}; \
-    done 
+    mkdir /projects/.che/gopath/bin -p && \
```
Comment (nickboldt) [inline devspaces-udi/Dockerfile:185]: do you need something like this?
RUN source $REMOTE_SOURCES_DIR/gopls/cachito.env

File: devspaces-udi/Dockerfile [hunk 2]
```diff
@@ -180,12 +182,12 @@ RUN \
     # Golang (Tech Preview)
     ########################################################################
     dnf -y -q install golang glibc-devel zlib-devel libstdc++ libstdc++-devel && \
-    mkdir -p /projects/.che/gopath /.cache ${HOME}/go && \
-    tar -xvf /tmp/assets/asset-golang-$(uname -m).tar.gz --strip-components=1 -C /projects/.che/gopath/ && \
-    for f in "/home/user" "/etc/passwd" "/etc/group" "/projects" "/.cache" "/usr/share/gocode" "/opt/app-root/src/"; do \
```
Comment (nickboldt) [inline devspaces-udi/Dockerfile:185]: the first few dirs are updated elsewhere but do we need to chrgrp/chmod these dirs?

`"/.cache" "/usr/share/gocode" "/opt/app-root/src/"`

Comment (mkuznyetsov) [reply devspaces-udi/Dockerfile:185]: I ran the sample project on DevSpaces 3.1 without these chmod/chgrps, soo I reckon they're not needed?

**Review by svor — APPROVED** [review]
LGTM

**Pull request merged**